### DISCUSSION TITLE
Error while using proxy: ssl3_get_record:wrong version number

### DISCUSSION THREAD

**Discussion by NPhMKgbDNy1M** [renovatebot/renovate#15517] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Renovate 32.45.0 (docker image)

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab Community Edition 14.8.3

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

In my company we are using renovate with HTTP_PROXY/HTTPS_PROXY vars set to `http://proxy[REDACTED]:7070`.
If renovate tries to load a page which is not in NO_PROXY (like github.com), there is an SSL error. Curl with the same bearer token works (also tested in the Gitlab ci). Where does this error come from?

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Unknown GitHub error (repository=[REDACTED]/examples/maven-backend-example)
       "err": {
         "name": "RequestError",
         "code": "EPROTO",
         "timings": {
           "start": 1650892243798,
           "socket": 1650892243855,
           "lookup": 1650892243855,
           "connect": 1650892243855,
           "upload": 1650892243856,
           "error": 1650892243857,
           "phases": {"wait": 57, "dns": 0, "tcp": 0, "request": 1, "total": 59}
         },
         "message": "write EPROTO 140713797236608:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:\n",
         "stack": "RequestError: write EPROTO 140713797236608:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:\n\n    at ClientRequest.<anonymous> (/usr/src/app/node_modules/got/dist/source/core/index.js:962:111)\n    at Object.onceWrapper (events.js:520:26)\n    at ClientRequest.emit (events.js:412:35)\n    at ClientRequest.emit (domain.js:475:12)\n    at ClientRequest.origin.emit (/usr/src/app/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)\n    at TLSSocket.socketErrorListener (_http_client.js:475:9)\n    at TLSSocket.emit (events.js:412:35)\n    at TLSSocket.emit (domain.js:475:12)\n    at emitErrorNT (internal/streams/destroy.js:106:8)\n    at emitErrorCloseNT (internal/streams/destroy.js:74:3)\n    at processTicksAndRejections (internal/process/task_queues.js:82:21)\n    at WriteWrap.onWriteComplete [as oncomplete] (internal/stream_base_commons.js:94:16)",
         "options": {
           "headers": {
             "user-agent": "RenovateBot/32.31.1 (https://github.com/renovatebot/renovate)",
             "accept": "application/vnd.github.v3+json",
             "authorization": "***********",
             "accept-encoding": "gzip, deflate, br"
           },
           "url": "https://api.github.com/repos/spring-projects/spring-boot/tags?per_page=100",
           "hostType": "github-changelog",
           "username": "",
           "password": "",
           "method": "GET",
           "http2": false
         }
       }

```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
Please create only discussions in future.

Most likely you have a man in the middle device such as a DPI device interfering with the SSL. You probably need a custom certificate in the Renovate image so that it lets requests get exploited by such a device. The question is why you bypass your proxy if you have another device in the network anyway.

**Reply by NPhMKgbDNy1M** [reply]
Thank you for your answer. TLS/SSL is not touched on the proxy.

**Reply by philippe-granet** [reply]
@NPhMKgbDNy1M Same problem as you, it works if I disable HTTP2 ([enableHttp2](https://docs.renovatebot.com/configuration-options/#enablehttp2))

**Reply by NPhMKgbDNy1M** [reply]
@philippe-granet For you it was enabled and disabling it helped? I have not activated it, since default is false.

**Reply by philippe-granet** [reply]
> @philippe-granet For you it was enabled and disabling it helped? I have not activated it, since default is false.

Yes it was enabled, and disabling it fix the error

**Reply by MindTooth** [reply]
@NPhMKgbDNy1M Were you able to fix this?  I've tried for days, but no go.

**Reply by NPhMKgbDNy1M** [reply]
Unfortunately no

**Reply by MindTooth** [reply]
I wan to report back that I was able to fix my problem.  But, I needed to set the full FQDN and port for the address (internal Docker Hub proxy).  Not sure that this is a thing, but try to be as explicit you can with the address that you want to configure and shave off parts to see if you can find your limit.

**Reply by NPhMKgbDNy1M** [reply]
When I have a free minute, I see if that helps. Thanks!

**Reply by MindTooth** [reply]
I actually was able to clean my `no_proxy`.  If you prepend a period (`.`) to the domain, is seems to work. 🤷🏻‍♂️ 

```diff
- no_proxy=<domain>
+ no_proxy=.<domain>
```

Ref.: https://www.gnu.org/software/wget/manual/html_node/Proxies.html

**Comment by NPhMKgbDNy1M** [comment]
I think I know now, where this issue is coming from: https://github.com/gajus/global-agent/pull/61
global-agent got the last update two years ago. I don't think, this will get merged soon.
Can we fix this bug in Renovate? @rarkins 

Log using `renovate/renovate:35-slim:`

```json
{"context":{"package":"global-agent","namespace":"Agent","logLevel":10,"destination":"https://index.docker.io/v2/renovate/renovate/tags/list?n=10000","proxy":"http://[proxy]:7070/","requestId":0},"message":"proxying request","sequence":27,"time":1682008104970,"version":"1.0.0"}
{"context":{"package":"global-agent","namespace":"Agent","logLevel":10,"target":{"host":"index.docker.io","port":443,"proxy":{"authorization":null,"hostname":"[proxy]","port":7070},"tls":{"servername":"index.docker.io"}}},"message":"connecting","sequence":31,"time":1682008105117,"version":"1.0.0"}
{"context":{"package":"global-agent","namespace":"Agent","logLevel":20},"message":"created socket","sequence":32,"time":1682008105118,"version":"1.0.0"}
{"context":{"package":"global-agent","namespace":"Agent","logLevel":50,"error":{"errno":-71,"code":"EPROTO","syscall":"write","name":"Error","message":"write EPROTO C057E0C6107F0000:error:0A00010B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:355:\n","stack":"Error: write EPROTO C057E0C6107F0000:error:0A00010B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:355:\n\n    at WriteWrap.onWriteComplete [as oncomplete] (node:internal/stream_base_commons:94:16)"}},"message":"socket error","sequence":33,"time":1682008105121,"version":"1.0.0"}
```
